pkg/pipeline/BulkUpdateService.go

```diff
@@ -0,0 +1,248 @@
+	environmentRepository     cluster.EnvironmentRepository
+	pipelineRepository        pipelineConfig.PipelineRepository
+	appLevelMetricsRepository repository3.AppLevelMetricsRepository
+	client                    *http.Client
+}
+
+func NewBulkUpdateServiceImpl(bulkUpdateRepository bulkUpdate.BulkUpdateRepository,
+	chartRepository chartConfig.ChartRepository,
+	logger *zap.SugaredLogger,
+	chartTemplateService util.ChartTemplateService,
+	repoRepository chartConfig.ChartRepoRepository,
+	pipelineGroupRepository pipelineConfig.AppRepository,
+	refChartDir RefChartDir,
+	defaultChart DefaultChart,
+	mergeUtil util.MergeUtil,
+	repositoryService repository.ServiceClient,
+	chartRefRepository chartConfig.ChartRefRepository,
+	envOverrideRepository chartConfig.EnvConfigOverrideRepository,
+	pipelineConfigRepository chartConfig.PipelineConfigRepository,
+	configMapRepository chartConfig.ConfigMapRepository,
+	environmentRepository cluster.EnvironmentRepository,
+	pipelineRepository pipelineConfig.PipelineRepository,
+	appLevelMetricsRepository repository3.AppLevelMetricsRepository,
+	client *http.Client,
+) *BulkUpdateServiceImpl {
+	return &BulkUpdateServiceImpl{
+		bulkUpdateRepository:      bulkUpdateRepository,
+		chartRepository:           chartRepository,
+		logger:                    logger,
+		chartTemplateService:      chartTemplateService,
+		repoRepository:            repoRepository,
+		pipelineGroupRepository:   pipelineGroupRepository,
+		mergeUtil:                 mergeUtil,
+		refChartDir:               refChartDir,
+		defaultChart:              defaultChart,
+		repositoryService:         repositoryService,
+		chartRefRepository:        chartRefRepository,
+		envOverrideRepository:     envOverrideRepository,
+		pipelineConfigRepository:  pipelineConfigRepository,
+		configMapRepository:       configMapRepository,
+		environmentRepository:     environmentRepository,
+		pipelineRepository:        pipelineRepository,
+		appLevelMetricsRepository: appLevelMetricsRepository,
+		client:                    client,
+	}
+}
+func (impl BulkUpdateServiceImpl) FindBulkUpdateReadme(operation string) (BulkUpdateResponse, error) {
+	bulkUpdateReadme, err := impl.bulkUpdateRepository.FindBulkUpdateReadme(operation)
+	response := BulkUpdateResponse{}
+	if err != nil {
+		impl.logger.Errorw("error in fetching batch operation example", "err", err)
+		return response, err
+	}
+	script := BulkUpdateScript{}
+	err = json.Unmarshal([]byte(bulkUpdateReadme.Script), &script)
+	if err != nil {
+		impl.logger.Errorw("error in script value(in db) of batch operation example", "err", err)
+		return response, err
+	}
+	response = BulkUpdateResponse{
+		Operation: bulkUpdateReadme.Operation,
+		Script:    script,
+		ReadMe:    bulkUpdateReadme.Readme,
+	}
+	return response, nil
+}
+func (impl BulkUpdateServiceImpl) GetBulkAppName(bulkUpdatePayload BulkUpdatePayload) ([]*ImpactedObjectsResponse, error) {
+	impactedObjectsResponse := []*ImpactedObjectsResponse{}
```

Comment:
we use  
var impactedObjectsResponse []*ImpactedObjectsResponse
not
impactedObjectsResponse := []*ImpactedObjectsResponse{}

Reply:
This was changed for sending empty array response instead of null, should I change this?

Reply:
let it be then